src/magic_folder/test/test_web.py

```diff
@@ -654,3 +673,518 @@ def test_create_snapshot_fails(self, author, folder_name, path_outside_folder, s
+                treq,
+                AUTH_TOKEN,
+                b"POST",
+                self.url.child(folder_name),
+                dumps({
+                    "author": {"name": "kelly"},
+                    "personal_dmd": personal_dmd,
+                }).encode("utf8")
+            ),
+            succeeded(
+                AfterPreprocessing(
+                    lambda response: response.json().result,
+                    Equals({})
+                )
+            )
+        )
+
+        # confirm that the "list participants" API includes the added
+        # participant
+        self.assertThat(
+            authorized_request(
+                treq,
+                AUTH_TOKEN,
+                b"GET",
+                self.url.child(folder_name),
+            ),
+            succeeded(
+                AfterPreprocessing(
+                    lambda response: response.json().result,
+                    Equals({
+                        u"iris": {
+                            u"personal_dmd": folder_config["upload-dircap"],
+                        },
+                        u'kelly': {
+                            u'personal_dmd': personal_dmd,
+                        }
+                    })
+                )
+            )
+        )
+
+    @given(
+        author_names(),
+        folder_names(),
+    )
+    def test_add_participant_wrong_json(self, author, folder_name):
+        """
+        Missing keys in 'participant' JSON produces error
+        """
+        local_path = FilePath(self.mktemp())
+        local_path.makedirs()
+        folder_config = magic_folder_config(
+            create_local_author(author),
+            FilePath(self.mktemp()),
+            local_path,
+        )
+
+        root = create_fake_tahoe_root()
+        # put our Collective DMD into the fake root
+        root._uri.data[folder_config["collective-dircap"]] = dumps([
+            u"dirnode",
+            {
+                u"children": {
+                    author: format_filenode(folder_config["upload-dircap"]),
+                },
+            },
+        ])
+        tahoe_client = create_tahoe_client(
+            DecodedURL.from_text(u"http://invalid./"),
+            create_tahoe_treq_client(root),
+        )
+        treq = treq_for_folders(
+            Clock(),
+            FilePath(self.mktemp()),
+            AUTH_TOKEN,
+            {
+                folder_name: folder_config,
+            },
+            start_folder_services=False,
+            tahoe_client=tahoe_client,
+        )
+
+        # add a participant using the API
+        self.assertThat(
+            authorized_request(
+                treq,
+                AUTH_TOKEN,
+                b"POST",
+                self.url.child(folder_name),
+                dumps({
+                    "not-the-author": {"name": "kelly"},
+                }).encode("utf8")
+            ),
+            succeeded(
+                AfterPreprocessing(
```

Comment:
We should probably be checking the http response code in all these tests.

Reply:
cb32a91f (testing to see if github links to the commit automagically)